chpl/chpl-service/src/main/java/gov/healthit/chpl/validation/listing/reviewer/edition2015/RequiredCriteriaValidator.java

```diff
@@ -0,0 +1,57 @@
+package gov.healthit.chpl.validation.listing.reviewer.edition2015;
+
+import java.util.Date;
+import java.util.Optional;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+import gov.healthit.chpl.domain.CertificationResult;
+import gov.healthit.chpl.domain.CertifiedProductSearchDetails;
+import gov.healthit.chpl.util.ErrorMessageUtil;
+import gov.healthit.chpl.validation.listing.reviewer.Reviewer;
+
+@Component("requiredCriteriaValidator")
+public class RequiredCriteriaValidator implements Reviewer {
+
+    @Value("${criterion.170_315_d_12}")
+    private Integer criteriaD12Id;
+
+    @Value("${criterion.170_315_d_13}")
+    private Integer criteriaD13Id;
+
+    @Value("#{new java.text.SimpleDateFormat(\"MM/dd/yyyy\").parse(\"${cures.ruleEffectiveDate}\")}")
+    private Date ruleEffectiveDate;
+
+    private ErrorMessageUtil msgUtil;
+
+    @Autowired
+    public RequiredCriteriaValidator(ErrorMessageUtil msgUtil) {
+        this.msgUtil = msgUtil;
+    }
+
+    @Override
+    public void review(CertifiedProductSearchDetails listing) {
+        // These criterion are only required after effective rule date
```

Comment:
They're not allowed to be added until after ERD, so I don't know that we need to complicate the logic here